client/actions/planning/api.ts

```diff
@@ -115,36 +117,133 @@ function query(
                         return Promise.resolve(result);
                     }
 
-                    return Promise.resolve(response._items);
+                    return Promise.resolve(response);
                 } else {
                     return Promise.reject('Failed to retrieve items');
                 }
             }, (error) => Promise.reject(error))
     );
 }
 
+const handleItemsForLastFetchedDay = (
+    items: Array<IPlanningItem>,
+    params: IPlanningSearchParams = {},
+    total: number,
+    dispatch: any,
+    getState: () => any,
+): Promise<Array<IPlanningItem>> => {
+    if (items.length < 1) {
+        return Promise.resolve([]);
+    }
+
+    const lastDayGroupItems = selectors.planning.lastDayGroup(getState()) ?? [];
+    const itemIdsInList = selectors.planning.planIdsInList(getState());
+
+    const hasFetchedTotalItems = total === (items ?? []).length + (lastDayGroupItems ?? []).length;
+    const hasLoadedAllItems = total === itemIdsInList.length;
+
+    if (hasFetchedTotalItems || hasLoadedAllItems) {
+        dispatch(storeLastDayGroup([]));
+
+        const storedPlannings = selectors.planning.storedPlannings(getState());
+        const listItems: Array<IPlanningItem> = (itemIdsInList ?? []).map((id) => storedPlannings[id]);
+
+        // listItems might contain some or all of items, get unique set of all
+        const allItems = uniqBy([
+            ...listItems,
+            ...items,
+            ...lastDayGroupItems
+        ], (item) => item._id);
+
+        return Promise.resolve(allItems);
```

Comment:
I wonder why it should return also items which are already in the list? afaik that combining is happening later in reducer. here it should only return `items + lastDayGroupItems`, not any items fetched before